### PR TITLE
Feature/task 7

### PR DESCRIPTION
**Pull request by TaisiaKishynskaya**

### CODE REVIEW

**Comment by TaisiaKishynskaya** [comment]
It is 10 variant in task 7.

File: main.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+from task_7 import Instructor
+
+
+def main():
+    instructors = [
+        Instructor("Doe", "John", {}),
```
Comment (vloooo) [inline main.py]: single quotes

File: main.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        instructor.print_subjects_per_semester()  # List of subjects per semester
+
+        # Add a semester with user input
+        instructor.add_semester_with_subjects()
+
+        # Remove a semester
+        instructor.remove_semester(2)
+
+        # Remove subjects from a semester with user input
+        semester_num = int(input("Enter the semester number to remove subjects from: "))
```
Comment (vloooo) [inline main.py]: validate

File: task_7.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
+class Instructor:
+    def __init__(self, last_name, first_name, subjects_per_semester):
+        self.last_name = last_name
```
Comment (vloooo) [inline task_7.py]: add access modifiers

File: task_7.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    def add_subject_to_semester(self, subject, semester):
+        subjects = self.subjects_per_semester.get(semester, [])
+        subjects.append(subject)
+        self.subjects_per_semester[semester] = subjects
+
+    def remove_subject_from_semester(self, subject, semester):
+        subjects = self.subjects_per_semester.get(semester, [])
+        if subject in subjects:
+            subjects.remove(subject)
+            self.subjects_per_semester[semester] = subjects
```
Comment (vloooo) [inline task_7.py]:
```suggestion
            self.subjects_per_semester[semester].remove(subject)
```

File: task_7.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        else:
+            raise ValueError(f"{subject} not found in semester {semester}")
```
Comment (vloooo) [inline task_7.py]: single quotes

File: task_7.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+            raise ValueError(f"Semester {semester} already exists")
+        self.subjects_per_semester[semester] = subjects
+
+    def remove_semester(self, semester):
+        if semester not in self.subjects_per_semester:
+            raise ValueError(f"Semester {semester} does not exist")
+        del self.subjects_per_semester[semester]
+
+    def print_semester_subjects(self, semester):
+        subjects = self.subjects_per_semester.get(semester, [])
```
Comment (vloooo) [inline task_7.py]: add property

File: task_7.py [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+        print(f"Average number of subjects per semester: {num_subjects / num_semesters}")
+
+    def print_semester_with_most_subjects(self):
+        max_subjects = -1
+        max_semester = None
+        for semester, subjects in self.subjects_per_semester.items():
+            num_subjects = len(subjects)
+            if num_subjects > max_subjects:
+                max_subjects = num_subjects
+                max_semester = semester
```
Comment (vloooo) [inline task_7.py]: max(iterable, key=......)

File: task_7.py [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+    def add_semester_with_subjects(self):
+        semester = input("Enter a semester number: ")
+        try:
+            semester = int(semester)
+        except ValueError:
+            print("Invalid semester number. Please enter an integer.")
+            return
+        if semester in self.subjects_per_semester:
+            print(f"Semester {semester} already exists. Adding subjects to existing semester.")
```
Comment (vloooo) [inline task_7.py]: move to else

**Pull request closed without merging**